### PR TITLE
Randoms: Enable Lilligant to get Own Tempo

### PR DESCRIPTION
**Pull request by SolarisFox**
Nobody likes the current Pyroar set. :c

### CODE REVIEW

**Comment by TheImmortal** [comment]
Yeah, no. It's much better than your set. Have you seen its Attack stat? Change it back.

The Lilligant is a good change! I don't think it's necessary to comment there tbh.

**Comment by SolarisFox** [comment]
Bulldoze is for the speed drop on switches to pick a better follow-up move where the opponent would have otherwise outsped. I see it used occasionally on the NU ladder and it seems to be a decent set (as opposed to solarbeam which I have yet to see ever outside of a dedicated sun team). More importantly though, having Sunny Day forces it to get Heat Rock 100% of the time, which is pretty terrible for pyroar since it very infrequently gets a chance to set up Sunny Day anyways, and even if it does, it really only needs it the next turn assuming you're predicting the rock/water switch in, which you are just as well off hitting with 2 HP Grass hits or WoW anyways. I honestly don't think the moves are any better so much as taking Sunny Day off of Pyroar (or at least Heat Rock somehow) would improve the set. I think in all the randbats matches I've played I've seen Sunny Day used twice.

**Comment by TheImmortal** [comment]
I didn't realize it gets it 100% of the time, seeing as how it has five moves. This is what you can do to give it a chance of getting Life Orb. Add Hidden Power Grass to its original moves and add this check (at line 1320):

```
case 'hiddenpower':
	if (this.getType(move.type) === 'Grass' && hasMove['solarbeam']) rejected = true;
	break;
```

**Comment by SolarisFox** [comment]
Okay, sounds good. Updated.

File: data/scripts.js [hunk 1]
```diff
@@ -1317,6 +1317,9 @@ exports.BattleScripts = {
 				case 'explosion':
 					if (counter.setupType || hasMove['wish']) rejected = true;
 					break;
+				case 'hiddenpower':
+					if (this.getType(move.type) === 'Grass' && hasMove['solarbeam']) rejected = true;
```
Comment (ascriptmaster) [inline data/scripts.js]: Maybe check for both Solarbeam and Sunny Day?

Comment (ascriptmaster) [inline data/scripts.js]: Actually, Sunny Day / Fire Blast / Hyper Voice / HP Grass is now a real possibility and probably shouldn't be, and we should do something about that.

Comment (ascriptmaster) [inline data/scripts.js]: \*sigh\* `(this.getType(move.type) === 'Grass' && hasMove['solarbeam'] || hasMove['sunnyday'])` it is, then.

Comment (TheImmortal) [inline data/scripts.js]: Let me generate some sets and see what's the best thing to do.

**Comment by ascriptmaster** [comment]
For the record, in case it's useful:

    15:20	Ascriptmaster	!usage pyroar moves
    15:20	TIBot	Hyper Voice 95.306% | Dark Pulse 62.274% | Hidden Power Grass 56.992% | Fire Blast 53.223% | Flamethrower 50.621% | Overheat 23.170% | Will-O-Wisp 14.011% | Hidden Power Ice 6.516% | Taunt 6.286% | Hyper Beam 4.023% | Wild Charge 2.933% | Return 2.662% | Hidden Power Electric 2.351% | Other 19.634%

Sunny Day is basically seen nowhere on Pyroar (even Hyper Beam has higher usage, which probably says something)

**Comment by SolarisFox** [comment]
As far as I'm concerned, the only moves that matter on Pyroar are [fire attack] / Hyper Voice / [WoW or HP grass]. Pyroars move pool is downright terrible so it wants to just spam Fire Blast and Hyper Voice as much as it can, seeing as those are it's only good two moves. Whatever else is put on Pyroar is only there to discourage things that resist those moves (mostly rock types) from switching in and reducing how often Pyroar is able to spam it's good stabs. For the most part, Rock Types don't want to switch in on WoW or HP Grass so either of those moves fill out it's move pool about the best it can hope for.

I understand the philosophy of wanting Randbat Pokes to be able to deal with as many threats on their own as possible because there is no way to assure team synergy, but I think in this case Pyroar is actually being hurt by trying to give it more coverage. All people want to do with it is spam Fire Blast to revenge kill or at least cause serious damage to whatever switches in, and as such I think insuring that Pyroar always gets Life Orb Fire Blast + Hyper Voice is most important, and everything else on it's set should be a secondary consideration.

**Comment by TheImmortal** [comment]
There. You can remove the Pyroar changes and just keep this PR for Lilligant.

Now Pyroar gets Life Orb as much as it gets Heat Rock, and at the same time I improve Rapidash. Dark Pulse has a high usage in every tier for Pyroar so I chose that over Hidden Power Grass.

**Comment by SolarisFox** [comment]
Done & changed pullreq title

**Pull request merged**